### PR TITLE
OCPQE-22273: Add the missing permision for availabilitySets deletion

### PR DESCRIPTION
**Pull request by sunzhaohua2**
@jinyunma PTAL

### CODE REVIEW

File: ci-operator/step-registry/azure/provision/service-principal/minimal-permission/azure-provision-service-principal-minimal-permission-commands.sh [hunk 1]
```diff
@@ -200,6 +200,7 @@ ${required_permissions}
 if [[ "${CLUSTER_TYPE_MIN_PERMISSOIN}" == "IPI" ]]; then
     required_permissions="""
 \"Microsoft.Compute/availabilitySets/write\",
+\"Microsoft.Compute/availabilitySets/delete\",
```
Comment (jinyunma) [inline ci-operator/step-registry/azure/provision/service-principal/minimal-permission/azure-provision-service-principal-minimal-permission-commands.sh:203]: we don't need to add this permission on UPI?

Comment (sunzhaohua2) [reply ci-operator/step-registry/azure/provision/service-principal/minimal-permission/azure-provision-service-principal-minimal-permission-commands.sh:203]: Actually it's needed, I see write only added on IPI, so just add delete on IPI. On UPI ci, we can't create machineset automatically, so I think it's ok to only add it on IPI.
For the upi clusters, if we want to create/delete machineset manually, these permissions are needed.

**Comment by rioliu-rh** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-automated-release-azure-ipi-marketplace-mini-perm-f360

**Comment by sunzhaohua2** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-automated-release-azure-ipi-marketplace-mini-perm-f360

**Comment by sunzhaohua2** [comment]
The job passed in installation and the case OCP-25436 also passed  [link](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/52332/rehearse-52332-periodic-ci-openshift-openshift-tests-private-release-4.16-automated-release-azure-ipi-marketplace-mini-perm-f360/1793180572302970880)

**Pull request merged**